flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/ArcticCatalog.java

```diff
@@ -359,8 +349,184 @@ private static void validateFlinkTable(CatalogBaseTable table) {
 
   @Override
   public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean ignoreIfNotExists)
-      throws CatalogException {
-    throw new UnsupportedOperationException();
+      throws CatalogException, TableNotExistException {
+    validateFlinkTable(newTable);
+
+    TableIdentifier tableIdentifier = getTableIdentifier(tablePath);
+    ArcticTable arcticTable;
+    try {
+      arcticTable = internalCatalog.loadTable(tableIdentifier);
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
+      if (!ignoreIfNotExists) {
+        throw new TableNotExistException(internalCatalog.name(), tablePath, e);
+      } else {
+        return;
+      }
+    }
+
+    CatalogTable table = toCatalogTable(arcticTable, tableIdentifier);
+    // Currently, Flink SQL only support altering table properties.
+    validateTableSchemaAndPartition(table, (CatalogTable) newTable);
+
+    if (arcticTable.isUnkeyedTable()) {
+      alterUnKeyedTable(arcticTable.asUnkeyedTable(), newTable);
+    } else if (arcticTable.isKeyedTable()) {
+      alterKeyedTable(arcticTable.asKeyedTable(), newTable);
+    } else {
+      throw new UnsupportedOperationException("Unsupported alter table");
+    }
+  }
+
+  private void alterKeyedTable(KeyedTable table, CatalogBaseTable newTable) {
+    Map<String, String> oldProperties = table.properties();
+    Map<String, String> setProperties = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : newTable.getOptions().entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (!java.util.Objects.equals(value, oldProperties.get(key))) {
+        setProperties.put(key, value);
+      }
+    }
+    oldProperties
+        .keySet()
+        .forEach(
+            k -> {
+              if (!newTable.getOptions().containsKey(k)) {
+                setProperties.put(k, null);
+              }
+            });
+    commitKeyedChanges(table, setProperties);
+  }
+
+  private void commitKeyedChanges(KeyedTable table, Map<String, String> setProperties) {
+    if (!setProperties.isEmpty()) {
+      commitUnKeyedChanges(table.baseTable(), null, null, null, setProperties);
+      if (table.changeTable() != null) {
+        commitUnKeyedChanges(table.changeTable(), null, null, null, setProperties);
+      }
+    }
+  }
+
+  private void alterUnKeyedTable(UnkeyedTable table, CatalogBaseTable newTable) {
+    Map<String, String> oldProperties = table.properties();
```

Comment:
Perhaps we should reuse org.apache.iceberg.flink.FlinkCatalog  instead of rewriting the logic again.

Reply:
These methods are private in `org.apache.iceberg.Flink.FlinkCatalog` and I cannot reuse them.

Reply:
As far as i know org.apache.iceberg.flink.FlinkCatalog#alterTable is public

Reply:
How should I initialize `FlinkCatalog`? Does `amoro` have an API? I'm not very familiar, sorry.